backend/server.js

```diff
@@ -134,39 +180,85 @@ function addBackendRoutes (app, config, store) {
      grants. */
   app.post('/upload', function (req, res) {
     config.getUserConfig(req, function (err, userConfig) {
-      selectTarget (userConfig, req.body, function (err, target) {
-        if (err) return res.json({error: err.toString()});
+      selectTarget(userConfig, req.body, function (err, target) {
+        if (err) return res.json({ error: err.toString() });
         const s3client = upload.makeS3UploadClient(target);
-        const {s3Bucket, uploadPath: uploadDir} = target;
+        const { s3Bucket, uploadPath: uploadDir } = target;
+        const targetName = req.body.targetName;
         const id = Date.now().toString();
         const uploadPath = `${uploadDir}/${id}`;
         upload.getJsonUploadForm(s3client, s3Bucket, uploadPath, function (err, events) {
-          if (err) return res.json({error: err.toString()});
+          if (err) return res.json({ error: err.toString() });
           upload.getMp3UploadForm(s3client, s3Bucket, uploadPath, function (err, audio) {
-            if (err) return res.json({error: err.toString()});
+            if (err) return res.json({ error: err.toString() });
             const baseUrl = `https://${s3Bucket}.s3.amazonaws.com/${uploadPath}`;
             const player_url = `${config.playerUrl}?base=${encodeURIComponent(baseUrl)}`;
-            res.json({player_url, events, audio});
+            if (!config.enableOauth && config.database)
+              mysqlUtils.storeRecord(req.session.userId, targetName, baseUrl, id, config.database);
+            res.json({ player_url, events, audio });
           });
         });
       });
     });
   });
 
+  app.post('/delete', checkLogin, function (req, res) {
```

Comment:
Make that route `/record/delete` to make it more descriptive.

Reply:
done